pymc/logprob/utils.py

```diff
@@ -294,3 +305,32 @@ def reconsider_logprob(rv: TensorVariable) -> TensorVariable:
     new_node.op = copy(new_node.op)
     new_node.op.__class__ = original_op_type
     return new_node.outputs[node.outputs.index(rv)]
+
+
+def ignore_logprob_multiple_vars(
+    vars: Sequence[TensorVariable], rvs_to_values: Dict[TensorVariable, TensorVariable]
+) -> List[TensorVariable]:
+    """Return duplicated variables that are ignored when creating logprob graphs.
+
+    This function keeps any interdependencies between variables intact, after
+    making each "unmeasurable", whereas a sequential call to `ignore_logprob`
+    would not do this correctly.
```

Comment:
How would sequential calls to `ignore_logprob` not do this correctly? I see that you make use of [`_replace_rvs_in_graph`](https://github.com/pymc-devs/pymc/blob/main/pymc/pytensorf.py#L208) with `replacement_fn`, perhaps that's the magic but it's a bit unclear to me...

Reply:
If you have:

```python
x = pm.Normal.dist()
y = pm.Normal.dist(x)

new_x = ignore_logprob(x)
new_y = ignore_logprob(y)
assert new_y.owner.inputs[3] is new_x  # Would fail
```